In [23]:
import numpy as np
import matplotlib.pyplot as plt
from matplotlib.animation import FuncAnimation
from sympy import symbols, Matrix, pi, cos, sin, simplify, eye, solve, latex, atan2, pprint, init_printing, Derivative, sqrt, integrate, Eq
from sympy.physics.mechanics import ReferenceFrame, dynamicsymbols, init_vprinting

# Configuración de impresión en formato LaTeX (MathJax)
init_vprinting(use_latex='mathjax')

---------------------PUNTO 1------------------

In [24]:
#Las barras no se deforman, entonces su longitud es un simbolo
b  = symbols("b")
#Se nos dice que la barra A horizontal se esta moviendo con aceleracion constante a la derecha
aceleracion_A = symbols("a_A")
#El angulo de elevacion de CB (theta)
#El angulo de elevancion de BA
#Todas las 3 variables cambian con el tiempo, a medida que la barra se desliza por la derecha
theta, gamma = dynamicsymbols("theta gamma")
#Creamos el tiempo
t = symbols("t")
#La longitud X
x = integrate(integrate(aceleracion_A, symbols("t")), symbols("t")) 
#Plnateamos nuestro marco de referencia fijo en el punto C
N = ReferenceFrame("N")
#Planteamos nuestro marco de referencia movil en la revoluta C, que seguira a la barra
#El eje X de este marcp movil seguira a la barra CB
C = N.orientnew("C", "Axis", (theta, N.z))
#Planteamos nuestro marco de referencia movil en B, en la revoluta
#El eje X del eje movil seguira a la barra BA
B = N.orientnew("B", "Axis", (-(gamma), N.z)) #debe ser este valor pues gira en sentido horario (negativo) y si tomammos a gamma como el ángulo del triángulo de arriba, entonces -(pi - gamma) es el ángulo que forma la barra BA con la horizontal

#Planteamos el loop de vevtores
r1 = b*C.x #vector CB
r2 = b*B.x #vector BA
r3 = x*N.x #vector x
eq = r1 + r2 - r3 #ecuacion del loop de vectores
display(eq)

#Calculamos la derivada de la ecuacion del loop de vectores
#La cual es la ecuacion de las velocidades
eq_velocidades = eq.diff(t,N)
display(eq_velocidades)
#Calculamos derivada de nuevo para la aceleracion
eq_aceleraciones = eq_velocidades.diff(t,N) #de esta ecuacion, unicamente conocemos x doble punto, que es "a"
display(eq_aceleraciones)


#Ahora lo que nos queda es dividir las ecuaciones en dos y resolver
#Ecuaciones de posicion
eqPos = [eq.express(N).dot(N.x), eq.express(N).dot(N.y)]
#Resolvemos las ecuaciones de posicion
# Ecuaciones de posición
eqPos = [eq.express(N).dot(N.x), eq.express(N).dot(N.y)]

# Hallamos theta y gamma en función de x
sln = solve(eqPos, (theta, gamma), dict=True)
display(sln)
#Tomamos solo la solucion negativa
#Pues como definimos nuestro marco en negativo, entonces hará que en total sea positivo, que es lo esperado
sol_gamma = sln[0][gamma]
w_buscado = sol_gamma.diff(t)
display(w_buscado)


                      2     
                -a_A⋅t      
b c_x + b b_x + ──────── n_x
                   2        

(-a_A⋅t - b⋅sin(γ)⋅γ̇ - b⋅sin(θ)⋅θ̇) n_x + (-b⋅cos(γ)⋅γ̇ + b⋅cos(θ)⋅θ̇) n_y

⎛                                           2             2⎞       ⎛           ↪
⎝-a_A - b⋅sin(γ)⋅γ̈ - b⋅sin(θ)⋅θ̈ - b⋅cos(γ)⋅γ̇  - b⋅cos(θ)⋅θ̇ ⎠ n_x + ⎝b⋅sin(γ)⋅γ̇ ↪

↪ 2             2                          ⎞    
↪   - b⋅sin(θ)⋅θ̇  - b⋅cos(γ)⋅γ̈ + b⋅cos(θ)⋅θ̈⎠ n_y

⎡⎧         ⎛     2⎞                 ⎛     2⎞      ⎫  ⎧       ⎛     2⎞          ↪
⎢⎪         ⎜a_A⋅t ⎟                 ⎜a_A⋅t ⎟      ⎪  ⎪       ⎜a_A⋅t ⎟          ↪
⎢⎨γ: - acos⎜──────⎟ + 2⋅π, θ: - acos⎜──────⎟ + 2⋅π⎬, ⎨γ: acos⎜──────⎟, θ: acos ↪
⎢⎪         ⎝ 4⋅b  ⎠                 ⎝ 4⋅b  ⎠      ⎪  ⎪       ⎝ 4⋅b  ⎠          ↪
⎣⎩                                                ⎭  ⎩                         ↪

↪ ⎛     2⎞⎫⎤
↪ ⎜a_A⋅t ⎟⎪⎥
↪ ⎜──────⎟⎬⎥
↪ ⎝ 4⋅b  ⎠⎪⎥
↪         ⎭⎦

          a_A⋅t          
─────────────────────────
          _______________
         ╱      2  4     
        ╱    a_A ⋅t      
2⋅b⋅   ╱   - ─────── + 1 
      ╱           2      
    ╲╱        16⋅b       

En la solución del libro aparece que w tiene un valor en el numerador de raiz(2*a*x).
 Recordando que x = (1/2)*a*t^2 (integrar dos veces la aceleración), obtenemos que el numerador es

 w_numerador = raiz(2*a*((1/2)*a*t^2))

 w_numerador = raiz(((a*t)^2))

 w_numerador = a*t

 Que es la respuesta obtenida por nosotros


----------------------------

Ahora bien, en la solución del libro aparece que w tiene un valor en el denominador de raiz(4*b^2 - x^2)
Recordando el valor de x, tenemos que:

 w_denominador = raiz(4(b)^2 - x^2)

 w_denominador = raiz(4*(b)^2 - ((1/2)*a*t^2)^2)

 w_denominador = raiz((4*(b)^2 - (1/4)*(a^2)*t^4 )

 #factorizamos b^2
 w_denominador = raiz(b^2(4 - ((1/4)*(a^2)*t^4)/b^2 )

 w_denominador = b*raiz(4 - ((1/4)*(a^2)*t^4)/b^2 )
 #multiplicamos y dividimos por 2 (la multiplicacion queda con el b, la division se multiplica por la raiz)
 
 wdenominador = 2*b*(raiz(4 - (((1/4)*(a^2)*t^4)/b^2)))/2
 #el 1/2 se mete dentro de la raiz como 1/4 (raiz de 1/4 = 1/2)
 wdenominador = 2*b*(raiz(4 - (((1/4)*(a^2)*t^4)/b^2))/1/4 )
 
 wdenominador = 2*b*(raiz(4/4 - (((1/4)*(a^2)*t^4)/b^2)/4) )
 
 wdenominador = 2*b*(raiz(1 - (((1/16)*(a^2)*t^4)/b^2)) )


Que es la respuesta que se obtuvo.


---------------------PUNTO 2------------------

In [25]:
#Definimos distancias simbolicas
AB, BC  = symbols("AB BC")
#Definimos el angulo fijo de C
a_C = symbols("a_C")
#Definimos el angulo de elevancion theta de AB
#Tambien, definimos el angulo con respecto a la horizontal de BC
#Uno de llama theta, el otro gamma
theta, gamma = dynamicsymbols("theta gamma")
#Creamos el tiempo
t = symbols("t")

#Plnateamos nuestro marco de referencia fijo en el punto A
N = ReferenceFrame("N")
#Planteamos nuestro marco de referencia movil en la revoluta A, que seguira a la barra AB
#El eje X de este marcp movil seguira a la barra CB
A = N.orientnew("A", "Axis", (theta, N.z))
#Planteamos nuestro marco de referencia movil en B, en la revoluta
#El eje X del eje movil seguira a la barra CB
B = N.orientnew("B", "Axis", (gamma, N.z)) 
#Planteamos otro marco de referencia fijo: en C
#Este marco de referencia será el mismo que el original solo que rotado a_C grados
#Así, el vector entre C y A tendrá dos componenes: una horizontal que se encargará de desplazarse a lo largo del slider
#Y una vertica que será la encargada de acercarse a A
#La verical es constante
C = N.orientnew("C", "Axis", (a_C, N.z))
#Planteamos un simbolo para movernos por el slider de C
x = dynamicsymbols("x")


#Planteamos nuestro loop de vectores
r1 = AB*A.x #vector AB
r2 = -BC*B.x #vector BC
r3_x = -x*C.x #la componente horizontal puede iniciar en cualquier lado pero r3_y siempre es r3_y
#Definimos un simbolo h que represena a la magnitud que deberia tener r3
#r3_magnitud = (BC**(2)+(AB)**2-2*BC*AB*cos(pi-gamma-theta))**(1/2) #la explicacion de este calculo esta en el markdown de abajo
#Sin embargo, en las ecuaciones nos damos cuenta que para hallar la velocidad angular en realidad no es relevante este valor
h = symbols("h")
r3_y = -h*C.y #componente vertical del vector AC

#Establecemos la ecuacion del loop de vectores
eq = r1.express(N) + r2.express(N) + r3_x.express(N) + r3_y.express(N)
display(eq.simplify())
#Derivamos para calcular la ecuacion de las velocidades
eq_velocidades = eq.dt(N)
display(eq_velocidades.simplify())
#Resolvemos para WAB o theta punto
eqVel = [eq_velocidades.express(N).dot(N.x), eq_velocidades.express(N).dot(N.y)]
# Hallamos gamma punto en función de x
sol = solve(eqVel, (theta.diff(t), gamma.diff(t)), dict=True)
display(sol)


(AB⋅cos(θ) - BC⋅cos(γ) + h⋅sin(a_C) - x⋅cos(a_C)) n_x + (AB⋅sin(θ) - BC⋅sin(γ) ↪

↪  - h⋅cos(a_C) - x⋅sin(a_C)) n_y

(-AB⋅sin(θ)⋅θ̇ + BC⋅sin(γ)⋅γ̇ - cos(a_C)⋅ẋ) n_x + (AB⋅cos(θ)⋅θ̇ - BC⋅cos(γ)⋅γ̇ - s ↪

↪ in(a_C)⋅ẋ) n_y

⎡⎧              sin(a_C)⋅sin(θ)⋅ẋ                      cos(a_C)⋅cos(θ)⋅ẋ       ↪
⎢⎨γ̇: - ──────────────────────────────────── - ──────────────────────────────── ↪
⎣⎩     -BC⋅sin(γ)⋅cos(θ) + BC⋅sin(θ)⋅cos(γ)   -BC⋅sin(γ)⋅cos(θ) + BC⋅sin(θ)⋅co ↪

↪                     sin(a_C)⋅sin(γ)⋅ẋ                      cos(a_C)⋅cos(γ)⋅ẋ ↪
↪ ────, θ̇: - ──────────────────────────────────── - ────────────────────────── ↪
↪ s(γ)       -AB⋅sin(γ)⋅cos(θ) + AB⋅sin(θ)⋅cos(γ)   -AB⋅sin(γ)⋅cos(θ) + AB⋅sin ↪

↪           ⎫⎤
↪ ──────────⎬⎥
↪ (θ)⋅cos(γ)⎭⎦

In [26]:
#Ahora reemplazamos todoo en w_buscado
theta_punto_buscado = sol[0][theta.diff(t)]
diccti = {a_C:pi/4,
          theta:pi/3,
          x.diff(t):-8, #se mueve hacia abajo a la izquierda
          BC:0.35, #a metros
          gamma:0,
          AB:0.5, #a metros
          }
resultado = theta_punto_buscado.subs(diccti)
display(round(resultado,2), "radianes/sec")

13.06

'radianes/sec'

![Diagrama del mecanismo](fa98076d-8f39-4fbb-9708-e93801998a6f.jpg)


---------------------PUNTO 3------------------

In [27]:
#Definimos distancias simbolicas
AB, r  = symbols("AB r")

#Definimos el angulo de elevancion theta del centro a B
#Tambien, definimos el angulo con respecto a la horizontal de A
#Uno de llama theta, el otro gamma
theta, gamma = dynamicsymbols("theta gamma")
#Creamos el tiempo
t = symbols("t")

#Plnateamos nuestro marco de referencia fijo en el centro de la rueda
N = ReferenceFrame("N")
#Planteamos nuestro marco de referencia movil en la revoluta en el centro de la rueda, con su eje Y apuntando a él

#El eje y de este marcp movil seguira al extremo de la barra AB
#Orientamos el eje movil A hacia abajo ya que asi queda colineal con la punta de la barra
#Entonces debe ser -theta
#orientenew siempre lo mira antihorario
A = N.orientnew("A", "Axis", (-theta, N.z))
#Planteamos nuestro marco de referencia movil en B con respecto a A, en la revoluta
#Este sigue a la barra BA
#El eje X del eje movil seguira a la barra CB
B = N.orientnew("B", "Axis", (gamma, N.z)) 
#Planteamos un simbolo para movernos por el slider de A
x = dynamicsymbols("x")
#Como estan paralelos el marco de referencia hipotetico del Slider, podemos crear dos vectores como en el ejercicio anterior
#Uno sera r_3y que tendrá un valor constante h
h = symbols("h")
#Y otro será r_3x que tendrá un valor asociado a x

#Planteamos nuestro loop de vectores
r1 = -r*A.x #vector OB, debe ser negativo ya que A.x apunta hacia abajo derecha, y queremos que apunte hacia arriba izquierda.
r2 = AB*B.x #vector AB
r3_x = x*N.x #Movimiento del slider
r3_y = -h*N.y #debe ser negativa para ir de nuevo a la rueda


#Establecemos la ecuacion del loop de vectores
eq = r1.express(N) + r2.express(N) - r3_x.express(N) - r3_y.express(N)
display(eq.simplify())
#Derivamos para calcular la ecuacion de las velocidades
eq_velocidades = eq.dt(N)
display(eq_velocidades.simplify())
#Resolvemos para x punto y para gamma punto
eqVel = [eq_velocidades.express(N).dot(N.x), eq_velocidades.express(N).dot(N.y)]
# Hallamos gamma punto en función de x
sol = solve(eqVel, (x.diff(t), gamma.diff(t)), dict=True)
display(sol)

#Hacemos lo mismo para la aceleracion
#Lo unico es que nos toca aprovechar el calculo de gamma punto hallado en la velocidad cuando la utilicemos
eq_aceleraciones = eq_velocidades.dt(N)
eq_acel = [eq_aceleraciones.express(N).dot(N.x), eq_aceleraciones.express(N).dot(N.y)]
sol2 = solve(eq_acel, (x.diff(t,2), gamma.diff(t,2)), dict=True)
display(sol2)


(AB⋅cos(γ) - r⋅cos(θ) - x) n_x + (AB⋅sin(γ) + h + r⋅sin(θ)) n_y

(-AB⋅sin(γ)⋅γ̇ + r⋅sin(θ)⋅θ̇ - ẋ) n_x + (AB⋅cos(γ)⋅γ̇ + r⋅cos(θ)⋅θ̇) n_y

⎡⎧   -r⋅cos(θ)⋅θ̇      r⋅sin(γ)⋅cos(θ)⋅θ̇             ⎫⎤
⎢⎨γ̇: ────────────, ẋ: ───────────────── + r⋅sin(θ)⋅θ̇⎬⎥
⎣⎩    AB⋅cos(γ)            cos(γ)                   ⎭⎦

⎡⎧           2             2                          2     2                  ↪
⎢⎪   sin(γ)⋅γ̇    r⋅sin(θ)⋅θ̇    r⋅cos(θ)⋅θ̈       AB⋅sin (γ)⋅γ̇               2   ↪
⎢⎨γ̈: ───────── + ─────────── - ──────────, ẍ: - ───────────── - AB⋅cos(γ)⋅γ̇  - ↪
⎢⎪    cos(γ)      AB⋅cos(γ)    AB⋅cos(γ)           cos(γ)                      ↪
⎣⎩                                                                             ↪

↪                   2                                               ⎫⎤
↪  r⋅sin(γ)⋅sin(θ)⋅θ̇    r⋅sin(γ)⋅cos(θ)⋅θ̈                          2⎪⎥
↪  ────────────────── + ───────────────── + r⋅sin(θ)⋅θ̈ + r⋅cos(θ)⋅θ̇ ⎬⎥
↪        cos(γ)              cos(γ)                                 ⎪⎥
↪                                                                   ⎭⎦

In [28]:
#Ahora reemplazamos todo para la velocidad 
velocidad_slider = sol[0][x.diff(t)].simplify()
display(velocidad_slider)

#Hallamos la velocidad
dicti_velocidad = {
    r:0.15, #metros
    gamma:np.deg2rad(60),
    theta:np.deg2rad(30),
    theta.diff(t):8, #rad/s, dado en el enunciado
    AB:0.5, #metros
}
resultado = velocidad_slider.subs(dicti_velocidad)
display(round(resultado,2), "metros/sec")

r⋅sin(γ + θ)⋅θ̇
──────────────
    cos(γ)    

2.40

'metros/sec'

In [29]:
#Ahora reemplazamos todo para la aceleracion
aceleracion_slider = sol2[0][x.diff(t,2)].simplify()
display(aceleracion_slider)

eq_gamma_punto = sol[0][gamma.diff(t)].simplify()
display(eq_gamma_punto)

#calculamos la velocidad de gamma punto
valor_gamma_punto = eq_gamma_punto.subs(dicti_velocidad)

#Montamos el diccionario
dicti_aceleracion = {
    r:0.15, #metros
    gamma:np.deg2rad(60),
    gamma.diff(t):valor_gamma_punto,
    theta:np.deg2rad(30),
    theta.diff(t):8, #rad/s, dado en el enunciado
    theta.diff(t,2):16, #rad/s, dado en el enunciado
    AB:0.5, #metros
}
res = aceleracion_slider.subs(dicti_aceleracion)
display(res, "metros/sec^2")


      2                                  2
- AB⋅γ̇  + r⋅sin(γ + θ)⋅θ̈ + r⋅cos(γ + θ)⋅θ̇ 
──────────────────────────────────────────
                  cos(γ)                  

-r⋅cos(θ)⋅θ̇ 
────────────
 AB⋅cos(γ)  

-12.4800000000000

'metros/sec^2'

Por la forma como construimos al vector del slider (r3_x y r3_y) se suposo que su dirección iba hacia la derecha. Así, el valor de la velocidad positiva (2.4) lo confirma: La dirección de la velocidad es hacia el eje X positivo. Ahora bien, notamos una aceleracación es negativa: por lo que su dirección debe ser en Z negativo. Esto significa que se desplaza hacia la derecha, pero cada vez más lento.

---------------------PUNTO 4------------------

In [30]:
#Planteamos las variables que son funcion del tiempo
#Los thetas y x
theta1, theta2, x = dynamicsymbols("theta1 theta2 x")

#Establecemos las constantes longitudes de las barras
#Además de la distancia vertical con respecto a B entre O y B (LOAN de Longitud OB normal)
LOA, LAB, LOBN = symbols("LOA LAB LOBN")
#Creamos un simbolo para el angulo de elevacion de B
an_B = symbols("an_B")

#Establecemos el tiempo
t = symbols("t")

#Establecemos nuestro marco de referencia en O
N = ReferenceFrame("N")
#Nuestra revoluta A1 se encuentra rotando con respecto a la barra OA
A1 = N.orientnew("A1", "Axis", (theta1, N.z))
#Nuestra revoluta A2 hace referencia al punto A: diriendo a la barrra AB
A2 = N.orientnew("A2", "Axis", (-theta2, N.z))
#Nuestra revoluta B hace reference a la revoluta que está en el propio cilindro B
#La cual se encarga de conectar a O con B no mediante A1 y A2, sino mediante un vector directo escrito en este sistema de referencia
B = N.orientnew("B", "Axis", (an_B, N.z)) #Deberia tener un valor de -15 pues es el valor del ángulo de elevación del slider B.

#Creamos los vectores
r1 = LOA * A1.x #desde O hasta A
r2 = LAB * A2.x #desde A hasta B
r3_tangencial = -x * B.x  #desde B hasta O, debe ser negativo pues va en sentido contrario a B.x 
r3_normal = - LOBN*B.y #Análogo con la componente normal
r3 = r3_tangencial + r3_normal #unificamos ambos vectores en uno solo
#Establecemos la ecuacion del loop de vectores
eq = r1.express(N) +r2.express(N) +r3.express(N)

#Derivamos para calcular la ecuacion de las velocidades
eq_velocidades = eq.dt(N)

#Derivamos de nuevo para calcular la ecuacion de las aceleraciones
eq_acel = eq_velocidades.dt(N)

display(eq_acel)

#Ahora resolvemos esta ecuación con respecto a la aceleración del slider: x doble punto
#También buscamos que halle a theta2.diff(t,2) ya que se relaciona íntimamente con theta1 doble punto y x doble punto
ecuaciones_aceleracion = [eq_acel.dot(N.x), eq_acel.dot(N.y)]
resultad = solve(ecuaciones_aceleracion, x.diff(t,2), theta2.diff(t,2), dict=True)
display(resultad)

⎛                                2                                  2          ↪
⎝-LAB⋅sin(θ₂)⋅θ₂̈ - LAB⋅cos(θ₂)⋅θ₂̇  - LOA⋅sin(θ₁)⋅θ₁̈ - LOA⋅cos(θ₁)⋅θ₁̇  - cos(an ↪

↪      ⎞       ⎛              2                                  2             ↪
↪ _B)⋅ẍ⎠ n_x + ⎝LAB⋅sin(θ₂)⋅θ₂̇  - LAB⋅cos(θ₂)⋅θ₂̈ - LOA⋅sin(θ₁)⋅θ₁̇  + LOA⋅cos(θ ↪

↪                    ⎞    
↪ ₁)⋅θ₁̈ - sin(an_B)⋅ẍ⎠ n_y

⎡⎧                                      2                                      ↪
⎢⎪              LAB⋅sin(an_B)⋅cos(θ₂)⋅θ₂̇                         LAB⋅sin(θ₂)⋅c ↪
⎢⎨θ₂̈: ────────────────────────────────────────────── + ─────────────────────── ↪
⎢⎪    -LAB⋅sin(an_B)⋅sin(θ₂) + LAB⋅cos(an_B)⋅cos(θ₂)   -LAB⋅sin(an_B)⋅sin(θ₂)  ↪
⎣⎩                                                                             ↪

↪            2                                                                 ↪
↪ os(an_B)⋅θ₂̇                          LOA⋅sin(an_B)⋅sin(θ₁)⋅θ₁̈                ↪
↪ ─────────────────────── + ────────────────────────────────────────────── + ─ ↪
↪ + LAB⋅cos(an_B)⋅cos(θ₂)   -LAB⋅sin(an_B)⋅sin(θ₂) + LAB⋅cos(an_B)⋅cos(θ₂)   - ↪
↪                                                                              ↪

↪                                  2                                           ↪
↪          LOA⋅sin(an_B)⋅cos(θ₁)⋅θ₁̇                         LOA⋅sin(θ₁)⋅cos(an ↪
↪ ───────────────────

In [31]:
#Notamos que x doble punto está en función de
#LAB
#theta2
#theta2 punto
#anB
#LOA
#theta1
#theta1 punto
#theta1 doble punto

#Tenemos todo menos theta2 punto, que la podemos hallar con la ecuación de velocidades:
ecuaciones_velocidades = [eq_velocidades.dot(N.x), eq_velocidades.dot(N.y)]
sol_theta2_punto = solve(ecuaciones_velocidades, x.diff(t), theta2.diff(t), dict=True)[0][theta2.diff(t)]
display(sol_theta2_punto, "ecuacion theta2 punto")

#hallamos el valor de theta2 punto
#Reemplazamos los valores
dicti = {an_B:np.deg2rad(-15),
         LOA:0.075, # a metros
         LAB:0.225, #metros
         theta1:np.deg2rad(60), #en la imagen
         theta2:np.deg2rad(45), # en la imagen
         theta1.diff(t):10, #enunciado
         theta1.diff(t,2):0 #enunciado
        
}
valor_theta2_punto = sol_theta2_punto.subs(dicti)
display(valor_theta2_punto, "valor theta2 punto")

#Con este valor de theta2 punto ya podemos hallar x doble punto
dicti[theta2.diff(t)] = valor_theta2_punto

resultado_1 = resultad[0][x.diff(t,2)].subs(dicti)

display(resultado_1, "valor de x doble punto, aceleracion buscada")

#tambien hallamos theta2 doble punto
resultado_2 = resultad[0][theta2.diff(t,2)].subs(dicti)
display(resultado_2, "valor de la aceleracion de theta 2 doble punto")

           LOA⋅sin(an_B)⋅sin(θ₁)⋅θ₁̇                         LOA⋅cos(an_B)⋅cos( ↪
────────────────────────────────────────────── + ───────────────────────────── ↪
-LAB⋅sin(an_B)⋅sin(θ₂) + LAB⋅cos(an_B)⋅cos(θ₂)   -LAB⋅sin(an_B)⋅sin(θ₂) + LAB⋅ ↪

↪ θ₁)⋅θ₁̇           
↪ ─────────────────
↪ cos(an_B)⋅cos(θ₂)

'ecuacion theta2 punto'

0.996194969075615

'valor theta2 punto'

1.98360444978805

'valor de x doble punto, aceleracion buscada'

-36.6055374313868

'valor de la aceleracion de theta 2 doble punto'

En el libro aparece que la respuesta es 
alfa_AB = 36.6 rad/s^2 CCW
a_B = 1.984 m/s^2, 345°

Es claro que CCW iría en la dirección contraria a la que aumenta el ángulo: por lo que tiene sentido que el valor nos de negativo.

Nos coinciden ambos valores.

---------------------PUNTO 5------------------

In [32]:
#Definimos distancias simbolicas
OA, BO  = symbols("OA BO")

#Definimos el angulo de elevancia theta del de la horizontal con respecto a la barra OA theta
#Tambien, definimos el angulo con respecto a la horizontal de la depresión de la barra AB
#Uno se llama theta, el otro beta
#También, es necesario definir AB pues dado que es como un slider su longitud no es constante, varia en el tiempo
theta, beta = dynamicsymbols("theta beta")
#Creamos el tiempo
t = symbols("t")

#Plnateamos nuestro marco de referencia fijo en el centro de O
N = ReferenceFrame("N")
#Planteamos nuestro marco de referencia movil que se moverá con la barra AO
#El eje X de este marco se movera con la parra
O = N.orientnew("O", "Axis", (-theta, N.z))
#Planteamos nuestro marco de referencia movil en B  en la revoluta
#Este sigue a la barra BA
#El eje X del eje movil seguira a la barra BA
B = N.orientnew("B", "Axis", (beta, N.z)) 

#Podemos darle un valor a AB usando teorema del coseno
AB = (OA**2 + BO**2 - 2*OA*BO*cos(theta))**(0.5)
#Planteamos nuestro loop de vectores
r1 = -OA*O.x #vector AB, cambia su direccion, pero no su magnitud
r2 =-AB*B.x #vector BC, cambia su magnitud y direccion con betha
r3 = BO*N.x #valor fijo

#Establecemos la ecuacion del loop de vectores
eq = r1.express(N) + r2.express(N) + r3.express(N)

#Buscamos la solución de betha
eqs_pos = [eq.dot(N.x), eq.dot(N.y)]
sol_betha = solve(eqs_pos, (beta), dict=True) 
display(sol_betha)
#tomamos la solucion positiva
sol_betha = sol_betha[1][beta]

⎡⎧                          ⎛     0.707106781186548⋅OA⋅sin(θ)     ⎞⎫  ⎧        ↪
⎢⎪β: 3.14159265358979 - asin⎜─────────────────────────────────────⎟⎪, ⎪β: asin ↪
⎢⎨                          ⎜   __________________________________⎟⎬  ⎨        ↪
⎢⎪                          ⎜  ╱       2                        2 ⎟⎪  ⎪        ↪
⎣⎩                          ⎝╲╱  0.5⋅BO  - BO⋅OA⋅cos(θ) + 0.5⋅OA  ⎠⎭  ⎩        ↪

↪ ⎛     0.707106781186548⋅OA⋅sin(θ)     ⎞⎫⎤
↪ ⎜─────────────────────────────────────⎟⎪⎥
↪ ⎜   __________________________________⎟⎬⎥
↪ ⎜  ╱       2                        2 ⎟⎪⎥
↪ ⎝╲╱  0.5⋅BO  - BO⋅OA⋅cos(θ) + 0.5⋅OA  ⎠⎭⎦

In [33]:
#Derivamos para hallar betha punto
beta_dot = sol_betha.diff(t).simplify()
display(beta_dot)

#Hallamos el valor reemplazando
dicti = {
    OA: 2.5,
    BO:9,
    theta.diff(t): 120*2*pi/(60), #3.14 mejor que pi para que salga el resultado con números
}
sol_resultado = beta_dot.subs(dicti).simplify()
display(sol_resultado)


#Miramos si son iguales
sol_libro = 6.28*((cos(theta)-0.278)/(1.939-cos(theta)))
#digamos que theta vale 20|
dicti[theta] = np.deg2rad(20)
display(round(sol_resultado.subs(dicti),4))
display(round(sol_libro.subs(dicti),4))

                   ⎛                    2                                      ↪
1.4142135623731⋅OA⋅⎝0.353553390593274⋅BO ⋅cos(θ) + 0.353553390593274⋅BO⋅OA⋅sin ↪
────────────────────────────────────────────────────────────────────────────── ↪
                                    ______________________________________     ↪
                                   ╱                          2                ↪
                                  ╱           (BO - OA⋅cos(θ))             ⎛   ↪
                                 ╱   ──────────────────────────────────── ⋅⎝0. ↪
                                ╱          2                            2      ↪
                              ╲╱     0.5⋅BO  - 1.0⋅BO⋅OA⋅cos(θ) + 0.5⋅OA       ↪

↪ 2                                                    2       ⎞  
↪  (θ) - 0.707106781186548⋅BO⋅OA + 0.353553390593274⋅OA ⋅cos(θ)⎠⋅θ̇
↪ ────────────────────────────────────────────────────────────────
↪                                                                 
↪

                      ⎛                    2                                   ↪
0.00545342479538684⋅π⋅⎝7.95495128834866⋅sin (θ) + 30.8475333292631⋅cos(θ) - 15 ↪
────────────────────────────────────────────────────────────────────────────── ↪
            _____________________________________                              ↪
           ╱                                  2                                ↪
          ╱  -(0.277777777777778⋅cos(θ) - 1.0)                                 ↪
         ╱   ─────────────────────────────────── ⋅(1.0 - 0.515759312320917⋅cos ↪
       ╲╱           22.5⋅cos(θ) - 43.625                                       ↪

↪               ⎞
↪ .9099025766973⎠
↪ ───────────────
↪                
↪                
↪     3/2        
↪ (θ))           
↪                

4.1623

4.1583

En el libro aparece que la respuesta es
betha punto = 6.28 ((cos thetha - 0.278) / (1.939 - cos theta) ) rad/sec.

La expresión obtenida es supremamente compleja. Por tanto, si establecemos un valor de theta, digamos 20°, podemos saber si el resultado se acerca o no evaluando en cada uno. 

Al hacerlo, obtenemos resultados muy parecidos.